packages/davinci-client/src/lib/node.reducer.ts

```diff
@@ -194,5 +204,42 @@ export const nodeCollectorReducer = createReducer(initialCollectorValues, (build
         }
         return;
       }
+
+      if (collector.type === 'DeviceAuthenticationCollector') {
+        if (typeof action.payload.id !== 'string') {
+          throw new Error('Index argument must be a string');
```

Comment:
`TypeError`? What is the thought process in returning a error here versus our current idea of not returning errors?

Since we aren't catching this we are returning this to the user here, is this a new pattern?

Reply:
As mentioned on the call, this is not a new pattern, and the thrown error doesn't actually make it to the user. The original caller catches it, but I agree with your overall sentiment. We should revisit this and see if there's a better pattern.